library/nameops.ml

```diff
@@ -153,6 +153,9 @@ let name_max na1 na2 =
   | Name _ -> na1
   | Anonymous -> na2
 
+let rec pr_name_list ns =
+  Pp.prlist_with_sep spc (fun v -> Option.default (Pp.str "_") (Option.map pr_id v)) ns
+
```

Comment:
I find the name `pr_name_list` confusing since it is not about printing objects of type `Name.t` but about printing objects of type `Id.t option` (which is isomorphic but distinct).

I would rather call it `pr_id_opt_list`.

Actually, there is already a printer of objects of type `Name.t` which prints `_` for `Anonymous`. So, to avoid splitting the conventional way to print `Anonymous` over different parts of the code, the function `pr_name` should be reused.

But the problem is that we have here an `Id.t option` and not a `Name.t`. This suggests that what is wrong is the `VFun` constructor which should take a `Name.t` and not a `Id.t option`. By contagion, this means the same for `TacFun`, and, at the end, this is what I think would be the nicest way to proceed to avoid making the code more complicated than it is.

@tchajed: would you feel the courage to go this way, i.e. using `Name.t` rather than `Id.t option` in TacFun and VFun, to propagate the change accordingly, and then eventually use `pr_name` as it would be natural to do?